internal/mining/mining.go

```diff
@@ -62,6 +62,77 @@ type TxDesc struct {
 
 	// Fee is the total fee the transaction associated with the entry pays.
 	Fee int64
+
+	// NumSigOps is the total sigops for this transaction.
+	NumSigOps int
+
+	// NumP2SHSigOps is the total number of signature operations for all input
+	// transactions which are of the pay-to-script-hash type.
+	NumP2SHSigOps int
+}
+
+// TxAncestorStats is a descriptor that stores aggregated statistics for the
+// unconfirmed ancestors of a transasction.
+type TxAncestorStats struct {
+	// Fees is the sum of all fees of unconfirmed ancestors
+	// and the current transaction.
+	Fees int64
+
+	// SizeBytes is the total size of all unconfirmed ancestors.
+	SizeBytes int64
+
+	// NumSigOps is the total number of signature operations of all ancestors.
+	NumSigOps int64
+
+	// NumP2SHSigOps is the aggregate number of signature operations for all
+	// input transactions of all ancestors which are of the pay-to-script-hash
+	// type.
+	NumP2SHSigOps int64
+
+	// NumAncestors is the total number of ancestors for a given transaction.
+	NumAncestors int
+}
+
+// TxMiningView is a snapshot of the tx source.
+type TxMiningView interface {
+	// MiningDescs returns a slice of mining descriptors for all the
+	// transactions in the source pool.
+	TxDescs() []*TxDesc
+
+	// AncestorStats returns the last known bundle stats for a given transaction.
+	// May return a nil pointer if the stats have not been calculated. Calling
+	// Ancestors will calculate and update the value returned by this method.
+	AncestorStats(txHash *chainhash.Hash) *TxAncestorStats
+
+	// Ancestors returns all transactions in the mining view that the provided
+	// transaction has depends on.
+	Ancestors(txHash *chainhash.Hash) ([]*TxDesc, *TxAncestorStats)
+
+	// HasParents returns true if the provided transaction hash has any
+	// ancestors known to the view.
+	HasParents(txHash *chainhash.Hash) bool
+
+	// Parents returns a set of transactions in the graph that the provided
+	// transaction hash spends from in the view. The order of elements
+	// returned is not guaranteed.
+	Parents(txHash *chainhash.Hash) []*TxDesc
+
+	// Children returns a set of transactions in the graph that spend
+	// from the provided transaction hash. The order of elements
+	// returned is not guaranteed.
+	Children(txHash *chainhash.Hash) []*TxDesc
+
+	// Remove causes the provided transaction to be removed from the view, if
+	// it exists.
```

Comment:
I would suggest documenting what the `notifyFeeChange` parameter does.  What is it notifying?

Reply:
updated